src/coreComponents/functions/TableFunction.hpp

```diff
@@ -126,6 +126,21 @@ class TableFunction : public FunctionBase
     void
     compute( IN_ARRAY const & input, real64 & value, OUT_ARRAY && derivatives ) const;
 
+    /**
+     * @brief Move the KernelWrapper to the given execution space, optionally touching it.
+     * @param space the space to move the KernelWrapper to
+     * @param touch whether the KernelWrapper should be touched in the new space or not
+     * @note This function exists to enable holding KernelWrapper objects in an ArrayView
+     *       and have their contents properly moved between memory spaces.
+     */
+    void move( LvArray::MemorySpace const space, bool const touch = false )
```

Comment:
I copied the signature from `LvArray::ArrayView` just to be safe. I changed the default from `true` to `false` though, because for these objects it doesn't make sense to touch them, they're basically meant to be const in the kernel. 

In this case we can just drop the default, because it is only meant to be called by `ArrayView`, which always does it with both arguments.